Kafka/src/main/java/io/deephaven/kafka/publish/PublishToKafka.java

```diff
@@ -0,0 +1,138 @@
+package io.deephaven.kafka.publish;
+
+import io.deephaven.base.verify.Assert;
+import io.deephaven.db.tables.Table;
+import io.deephaven.db.v2.DynamicTable;
+import io.deephaven.db.v2.InstrumentedShiftAwareListenerAdapter;
+import io.deephaven.db.v2.StreamTableTools;
+import io.deephaven.db.v2.sources.chunk.Attributes;
+import io.deephaven.db.v2.sources.chunk.ObjectChunk;
+import io.deephaven.db.v2.sources.chunk.WritableObjectChunk;
+import io.deephaven.db.v2.utils.Index;
+import io.deephaven.db.v2.utils.OrderedKeys;
+import io.deephaven.db.v2.utils.ReadOnlyIndex;
+import io.deephaven.util.annotations.ReferentialIntegrity;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.jetbrains.annotations.NotNull;
+
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.KafkaProducer;
+
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+public class PublishToKafka<K, V> {
+    static final int CHUNK_SIZE = 2048;
+
+    @NotNull
+    private final Table table;
+    private final KafkaProducer<K, V> producer;
+    private final String topic;
+    private final KeyOrValueSerializer<K> keySerializer;
+    private final KeyOrValueSerializer<V> valueSerializer;
+
+    @ReferentialIntegrity
+    private final PublishListener publishListener;
+
+    public PublishToKafka(final Properties props,
+            final Table table,
+            final String topic,
+            final KeyOrValueSerializer<K> keySerializer,
+            final KeyOrValueSerializer<V> valueSerializer) {
+        this.table = table;
+        this.producer = new KafkaProducer<>(props);
+        this.topic = topic;
+        this.keySerializer = keySerializer;
+        this.valueSerializer = valueSerializer;
+
+
+        if (table.isLive()) {
+            ((DynamicTable) table.coalesce()).listenForUpdates(publishListener = new PublishListener());
+        } else {
+            publishListener = null;
+        }
+
+        // Publish the initial table state
+        publishMessages(table.getIndex(), false, true);
+    }
+
```

Comment:
We need a way to shutdown, which I think I need to use inside a callback-type object returned from `KafkaTools.produceFromTable`.  In my branch I am using the code below, not sure if that is right, but maybe.

```
    public PublishToKafka(final Properties props,
            final Table table,
            final String topic,
            final KeyOrValueSerializer<K> keySerializer,
            final KeyOrValueSerializer<V> valueSerializer) {
        this.producer = new KafkaProducer<>(props);
        this.topic = topic;
        this.keySerializer = keySerializer;
        this.valueSerializer = valueSerializer;


        if (table.isLive()) {
            final DynamicTable dynTable = ((DynamicTable) table.coalesce());
            this.table = dynTable;
            dynTable.listenForUpdates(publishListener = new PublishListener());
        } else {
            this.table = table;
            publishListener = null;
        }

        // Publish the initial table state
        publishMessages(table.getIndex(), false, true);
    }

    public void shutdown() {
        if (!table.isLive()) {
            return;
        }
        final DynamicTable dynTable = ((DynamicTable) table;
        dynTable.removeUpdateListener(publishListener);
    }
```